typescript/introduction/tsc-interfaces-classes/tsc-interfaces-intro.md

```diff
@@ -0,0 +1,113 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Introduction to Interfaces
+
+---
+## Content
+
+In TypeScript, the shape that represents a collection of key-value pairs is an object.
+
+One way to declare a type of an object is with `{}`:
+
+```ts
+function f(companyObj: { name: string }) {
+  console.log(companyObj.name);
+}
+const someObject = {
+  name: "Enki"
+};
+f(someObject);
+```
+
+It won't care about the order of the properties or if any additional ones exist.
+
+If we wanted to reuse the type of `companyObj` at multiple places in our program, we can extract it into an interface:
+
+```ts
+interface Company {
+  name: string;
+}
+
+function f(companyObj: Company) {
+  // ...
+}
+
+function g(otherCompanyObj: Company) {
+  // ...
+}
+```
+Now both `f` and `g` expect the same shape as their argument.
+
+Besides allowing us to define the shape of an object, interfaces let us declare properties as optional or read only:
+
+```ts
+interface Enki {
+  // an optional comment
+  comment?: string;
```

Comment:
this is not specific to interfaces - you can do the same on objects / with type aliases 

I think these properies like optional, readonly etc can be presented in the 3rd workout at the end